service/controller/clusterapi/v16/key/cluster_provider.go

```diff
@@ -0,0 +1,66 @@
+package key
+
+import (
+	"encoding/json"
+
+	g8sv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/cluster/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/runtime"
+	cmav1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+func clusterProviderSpec(cluster cmav1alpha1.Cluster) g8sv1alpha1.AWSClusterSpec {
+	spec, err := g8sClusterSpecFromCMAClusterSpec(cluster.Spec.ProviderSpec)
+	if err != nil {
+		panic(err)
```

Comment:
Why panic here? Is this for your testing? Otherwise I don't follow at all.

Reply:
It's the same as in `aws-operator`. They key functions' interface usability would be really bad with errors. If you check my resource implementation, I first check (with different function) whether the provider extension exists and cancel the resource if it doesn't. That way it won't panic during usage.